### PR TITLE
PXP-10565/PPS-177 fix: exclude file nodes from chart with options

### PR DESCRIPTION
**Pull request by mfshao**
Jira Ticket: [PXP-10565](https://ctds-planx.atlassian.net/browse/PXP-10565) [PXP-177](https://ctds-planx.atlassian.net/browse/PXP-177)

The current chart in homepage will automatically include file node counts in it. This PR add options to opt-out this auto-including feature

### New Features
- Homepage: added options to exclude File nodes from the charts

### Deployment changes
- New portal config options to exclude File nodes from the charts added: use `"graphql.chartNodesExcludeFiles"` to exclude File nodes from graphql chart, or use `"components.index.homepageChartNodesExcludeFiles"` to exclude File nodes from homepage chart. Note both of them are default to `false` so you have to explicitly set them to `true` to enable them

[PXP-10565]: https://ctds-planx.atlassian.net/browse/PXP-10565?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: data/getTexts.js [hunk 1]
```diff
@@ -6,11 +6,12 @@ const componentTexts = paramByApp(params, 'components');
 function getChartText() {
   const graphQL = getGraphQL(paramByApp(params, 'graphql'));
   const boardPluralNames = graphQL.boardCounts.map((item) => item.plural);
-  if (boardPluralNames.length < 4) { boardPluralNames.push('Files'); }
+  console.log(boardPluralNames);
```
Comment (paulineribeyre) [inline data/getTexts.js:9]: remove debug log?

File: docs/portal_config.md [hunk 2]
```diff
@@ -129,7 +132,7 @@ Below is an example, with inline comments describing what each JSON block config
           "name": "Documentation"
         }
       ],
-      "useProfileDropdown": false // optional; enables expiremental profile UI; defaults false, may change in future releases
+      "useProfileDropdown": false // optional; enables experimental profile UI; defaults false, may change in future releases
```
Comment (paulineribeyre) [inline docs/portal_config.md:135]: it's not experimental anymore, is it?

File: docs/portal_config.md [hunk 3]
```diff
@@ -78,6 +79,8 @@ Below is an example, with inline comments describing what each JSON block config
           "label": "Run analysis"
         }
       ],
+      "homepageChartNodesExcludeFiles": true, // optional; specifies whether to exclude File nodes from the homepage chart. Defaults to false
```
Comment (paulineribeyre) [inline docs/portal_config.md:82]: i think it's not in the right place, shouldn't it be under `components.index`?

Comment (paulineribeyre) [inline docs/portal_config.md:82]: we need to add `chartNodesExcludeFiles` to the `graphql` section too

**Review by paulineribeyre — COMMENTED** [review]
awesome!

File: docs/portal_config.md (hunk 2 above)
Comment (mfshao) [reply docs/portal_config.md:135]: well... it kind of still is, because only MIDRC is using this setup now 😅

File: docs/portal_config.md (hunk 3 above)
Comment (mfshao) [reply docs/portal_config.md:82]: the format of this file is a bit confusing, but this one is actually under `components.index` now

Comment (mfshao) [reply docs/portal_config.md:82]: it is there https://github.com/uc-cdis/data-portal/pull/1208/files#diff-b44409708be4e154876122765a9fd3ada56184bb299149d53f7227a5db35c2eaR46

Comment (paulineribeyre) [reply docs/portal_config.md:82]: oops, sorry

**Review by paulineribeyre — APPROVED** [review]

**Pull request merged**